code/modules/mob/living/basic/farm_animals/bee/_bee.dm

```diff
@@ -0,0 +1,338 @@
+#define BEE_POLLINATE_YIELD_CHANCE 33
+#define BEE_POLLINATE_PEST_CHANCE 33
+#define BEE_POLLINATE_POTENCY_CHANCE 50
+#define BEE_FOODGROUPS RAW | MEAT | GORE | BUGS
+
+/mob/living/basic/bee
+	name = "bee"
+	desc = "Buzzy buzzy bee, stingy sti- Ouch!"
+	icon_state = ""
+	icon_living = ""
+	icon = 'icons/mob/simple/bees.dmi'
+	gender = FEMALE
+	speak_emote = list("buzzes")
+
+	melee_damage_lower = 1
+	melee_damage_upper = 1
+	attack_verb_continuous = "stings"
+	attack_verb_simple = "sting"
+	response_help_continuous = "shoos"
+	response_help_simple = "shoo"
+	response_disarm_continuous = "swats away"
+	response_disarm_simple = "swat away"
+	response_harm_continuous = "squashes"
+	response_harm_simple = "squash"
+
+	mob_size = MOB_SIZE_LARGE
+	pixel_x = -16
+	base_pixel_x = -16
+
+	speed = 1
+	maxHealth = 10
+	health = 10
+	melee_damage_lower = 1
+	melee_damage_upper = 1
+	faction = list(FACTION_HOSTILE)
+	pass_flags = PASSTABLE | PASSGRILLE | PASSMOB | PASSMACHINE
+	mob_size = MOB_SIZE_TINY
+	mob_biotypes = MOB_ORGANIC|MOB_BUG
+	density = FALSE
+	gold_core_spawnable = FRIENDLY_SPAWN
+	can_be_held = TRUE
+	held_w_class = WEIGHT_CLASS_TINY
+	environment_smash  = ENVIRONMENT_SMASH_NONE
+	habitable_atmos = list("min_oxy" = 0, "max_oxy" = 0, "min_plas" = 0, "max_plas" = 0, "min_co2" = 0, "max_co2" = 0, "min_n2" = 0, "max_n2" = 0)
+	basic_mob_flags = DEL_ON_DEATH
+	ai_controller = /datum/ai_controller/basic_controller/bee
+	///he he he, beegent
+	var/datum/reagent/beegent = null
+	///the house we live in
+	var/obj/structure/beebox/beehome = null
+	///our icon base
+	var/icon_base = "bee"
+	var/isqueen = FALSE
+
+/mob/living/basic/bee/Initialize(mapload)
+	. = ..()
+	add_traits(list(TRAIT_SPACEWALK, TRAIT_VENTCRAWLER_ALWAYS), INNATE_TRAIT)
+	generate_bee_visuals()
+	AddElement(/datum/element/simple_flying)
+	AddComponent(/datum/component/clickbox, x_offset = -2, y_offset = -2)
+	AddComponent(/datum/component/swarming)
+	AddElement(/datum/element/swabable, CELL_LINE_TABLE_QUEEN_BEE, CELL_VIRUS_TABLE_GENERIC_MOB, 1, 5)
+	RegisterSignal(src, COMSIG_HOSTILE_POST_ATTACKINGTARGET, PROC_REF(apply_chemicals))
+
+/mob/living/basic/bee/mob_pickup(mob/living/picker)
+	if(flags_1 & HOLOGRAM_1)
+		return
+	var/obj/item/clothing/head/mob_holder/destructible/holder = new(get_turf(src), src, held_state, head_icon, held_lh, held_rh, worn_slot_flags)
+	var/list/reee = list(/datum/reagent/consumable/nutriment/vitamin = 5)
+	if(beegent)
+		reee[beegent.type] = 5
+	holder.AddComponent(/datum/component/edible, reee, null, BEE_FOODGROUPS, 10, 0, list("bee"), null, 10)
+	picker.visible_message(span_warning("[picker] scoops up [src]!"))
+	picker.put_in_hands(holder)
+
+/mob/living/basic/bee/death(gibbed)
+	if(beehome)
+		beehome.bees -= src
+		beehome = null
+	beegent = null
+	return ..()
+
+/mob/living/basic/bee/UnarmedAttack(atom/target, proximity_flag, list/modifiers)
```

Comment:
if we're going to override the whole UnarmedAttack proc I think we should use the signal instead. I know there was a bit of a discussion between Melbert and I in another PR but I think if we're going to be killing the entire proc anyhow (and not invoking parent at the start) there shouldn't really be a reason why we shouldn't do this on a signal... any concerns about this though? I don't know if the comsig for attack allows for return types (e.g. some sort of `COMPONENT_CANCEL_SIGNAL` thing)

Reply:
I don't think there's any immediately wrong with overriding `UnarmedAttack` since it's the highest level part of the click chain you can get on the mob without going directly into Click

(However you should check proximity flag, it's only set by telekinesis but you never know)

That being said apparently basic mobs do have `/melee_attack` which is called from `UnarmedAttack` which might be more appropriate to override-while-conditionally-calling-parent

Reply:
i would be amenable to either `melee_attack()` or a signal handler for this then, i just don't like completely glonking the click chain which is going to always be subject to a bunch of bullshit in the past, present, and future.